src/pages/TermsOfService.js

```diff
@@ -37,23 +39,37 @@ const TermsOfServicePage = () => {
     try {
       setBusy(true)
       const { enabled } = await Ajax().User.acceptTos()
-      termsOfService.userAcceptedTos = true
-      termsOfService.userAcceptedVersion = termsOfService.currentVersion
-      termsOfService.userNeedsToAcceptTos = false
-      authStore.update(state => ({ ...state, termsOfService }))
+
       if (enabled) {
+        termsOfService.userCanUseTerra = true
+        termsOfService.showTosPopup = false
+        termsOfService.userAcceptedVersion = termsOfService.currentVersion
+        termsOfService.userContinuedUnderGracePeriod = undefined
+
         const registrationStatus = userStatus.registeredWithTos
-        authStore.update(state => ({ ...state, registrationStatus }))
-      }
-      if (isGracePeriod) {
-        Nav.history.goBack()
+        authStore.update(state => ({ ...state, registrationStatus, termsOfService }))
+        Nav.goToPath('root')
+      } else {
+        reportError('Error accepting TOS, Sam accept TOS endpoint returned False')
```

Comment:
This message will be shown to users, so I'd suggest leaving out technical details such as service name and endpoint.